### PR TITLE
openapi: Generic runtime URLs

### PR DESCRIPTION
**Pull request by mitjat**
This PR makes runtimes' URLs generic, e.g. `/emerald/transaction` becomes `/{runtime}/transaction`.

`oapi-codegen` imports a symbol called `runtime` in the autogenerated code, which clashes with our URL param of the same name. The PR includes a bash script (called from `Makefile`) to work around this issue.

The PR also includes some general small cleanups -- I searched for "emerald" in the codebase and tweaked all mentions/comments that will soon apply to all runtimes, not just emerald.

Testing: No regressions on URLs from `https://github.com/oasisprotocol/oasis-indexer/blob/3c756580ea3fe1d7d5949b4062c30693074d8ad8/tests/e2e_regression/run.sh`.

### CODE REVIEW

File: go.mod
Comment (mitjat) [inline go.mod:24]: All the changes in go.sum and in this file are just due to a routine `go mod tidy`.

File: storage/migrations/02_oasis_3_runtimes.up.sql
Comment (mitjat) [inline storage/migrations/02_oasis_3_runtimes.up.sql:77]: This whole block is just moved from below, so that the file keeps its structure:
1) a section of tables that are (mostly) not tied to sdk-modules
2) one section per sdk module
3) permissions

File: analyzer/evmtokens/evm_tokens.go
Comment (aefhm) [inline analyzer/evmtokens/evm_tokens.go:29]: Should we rename this variable too?

File: analyzer/runtime/runtime.go
Comment (Andrew7234) [inline analyzer/runtime/runtime.go:206]: this is guaranteed to be the same for all current+future runtimes?

File: analyzer/evmtokens/evm_tokens.go
Comment (Andrew7234) [inline analyzer/evmtokens/evm_tokens.go:194]: `emerald` specifically? Will we use this for sapphire?

File: scripts/namespace_codegen_templates.sh
Comment (Andrew7234) [inline scripts/namespace_codegen_templates.sh:22]: 🤯 

(going to assume this works if `make oasis-indexer` works 😄 )

File: analyzer/evmtokens/evm_tokens.go
Comment (mitjat) [reply analyzer/evmtokens/evm_tokens.go:29]: Sure, lemme do that. I'll do it in my morning; can you give me a LGTM if things otherwise look OK?

File: analyzer/runtime/runtime.go
Comment (mitjat) [reply analyzer/runtime/runtime.go:206]: looks like it. Comes from [here](https://github.com/oasisprotocol/oasis-sdk/blob/b37e6da699df331f5a2ac62793f8be099c68469c/client-sdk/go/modules/rewards/types.go#L10-L10). I guess the original comment here was misguided.

File: analyzer/evmtokens/evm_tokens.go
Comment (mitjat) [reply analyzer/evmtokens/evm_tokens.go:194]: Not emerald specific, thanks. Lemme fix before merging the PR.

File: scripts/namespace_codegen_templates.sh
Comment (mitjat) [reply scripts/namespace_codegen_templates.sh:22]: bash is a dangerous weapon :)
it's not as unreadable as it looks at first glance; the commands are actually simply, just the syntax is ugly.

it does work; I don't much worry about it breaking since we're using a fixed version of the codegen. However :exploding_head: is :no_good:, and I seriously considered not doing this and renaming `runtime` instead. But it's such a prominently displayed, outward-facing variable in URLs, and there's not a good alternative. `rtime`, `rt` ... too cryptic, inconsistent with other names. `paratime` ... not used in code (and the name does penetrate our Go code in many places).

**Review by aefhm — APPROVED** [review]

File: scripts/namespace_codegen_templates.sh
Comment (Andrew7234) [reply scripts/namespace_codegen_templates.sh:22]: Makes sense, thanks for explaining the thought process

File: api/spec/v1.yaml
Comment (Andrew7234) [inline api/spec/v1.yaml:52]:
```suggestion
      The layer for which to return the requested data timeline.
```

**Review by Andrew7234 — APPROVED** [review]
Looks good!

File: api/spec/v1.yaml
Comment (mitjat) [reply api/spec/v1.yaml:52]: Uh bad copy paste. Rewrote from scratch. Thank you!

File: analyzer/runtime/runtime.go
Comment (pro-wh) [inline analyzer/runtime/runtime.go:139]: oh no, my precious copied boilerplate code!

**Pull request merged**
